iml-agent/src/cli.rs

```diff
@@ -43,8 +89,69 @@ fn run_cmd<R: Send + 'static, E: Send + 'static>(
     tokio::runtime::Runtime::new().unwrap().block_on_all(fut)
 }
 
+fn input_to_iter(input: Option<String>, fidlist: Vec<String>) -> Box<Iterator<Item = String>> {
+    match input {
+        None => {
+            if fidlist.is_empty() {
+                Box::new(
+                    BufReader::new(io::stdin())
+                        .lines()
+                        .map(|x| x.expect("Failed to readline from stdin")),
+                )
+            } else {
+                Box::new(fidlist.into_iter())
+            }
+        }
+        Some(name) => {
+            let buf: Box<BufRead> = match name.as_ref() {
+                "-" => Box::new(BufReader::new(io::stdin())),
+                _ => {
+                    let f = match File::open(&name) {
+                        Ok(x) => x,
+                        Err(e) => {
+                            log::error!("Failed to open {}: {}", &name, e);
+                            exit(-1);
+                        }
+                    };
+                    Box::new(BufReader::new(f))
+                }
+            };
+            Box::new(
+                buf.lines()
+                    .map(|x| x.expect("Failed to readline from file")),
+            )
+        }
+    }
+}
+
 fn main() {
     env_logger::init();
 
-    let _matches = App::from_args();
+    let matches = App::from_args();
+
+    match matches {
+        App::Action { command: cmd }=> match cmd {
+            ActionRunner::Purge{ fidopts: opt } => {
+                let device = opt.fsname;
+                let input = input_to_iter(opt.input, opt.fidlist);
+
+                if liblustreapi::rmfid(&device, input).is_err() {
+                    exit(-1);
+                }
+            },
+            ActionRunner::Warning{ output: out, fidopts: opt } => {
+                let device = opt.fsname;
+                let output: Box<io::Write> = match out {
+                    Some(file) => Box::new(File::create(file).expect("Failed to create file")),
+                    None => Box::new(io::stdout()),
+                };
+                let input = input_to_iter(opt.input, opt.fidlist);
+
+                if stratagem_action_warning::write_records(&device, input, output).is_err() {
+                    exit(-1);
```

Comment:
Use `exitcode` here:

https://rust-lang-nursery.github.io/cli-wg/in-depth/exit-code.html?highlight=exitcode#exit-codes